llvm/lib/CodeGen/LiveIntervals.cpp

```diff
@@ -730,7 +730,16 @@ void LiveIntervals::addKillFlags(const VirtRegMap *VRM) {
     // Find the regunit intervals for the assigned register. They may overlap
     // the virtual register live range, cancelling any kills.
     RU.clear();
-    for (MCRegUnit Unit : TRI->regunits(PhysReg)) {
+    LaneBitmask ArtificialLanes = LaneBitmask::getNone();
+    for (MCRegUnitMaskIterator UI(PhysReg, TRI); UI.isValid(); ++UI) {
+      auto [Unit, Bitmask] = *UI;
+      // Record lane mask for all artificial RegUnits for this physreg.
+      for (MCRegUnitRootIterator Root(Unit, TRI); Root.isValid(); ++Root) {
+        if (TRI->isArtificial(*Root)) {
+          ArtificialLanes |= Bitmask;
+          break;
+        }
+      }
```

Comment:
Move this to a helper function that returns the mask instead of breaking out of the loop

Reply:
Perhaps you reviewed an older revision of the PR, I had moved the check to a helper function in 5e5c20cfdb653ac7da2b17b46eab9dc694d558b7